include/hardware/Encoder/V5RotationSensor.hpp

```diff
@@ -13,30 +14,30 @@ class V5RotationSensor : public Encoder {
         /**
          * @brief Construct a new V5 Rotation Sensor
          *
-         * @param encoder the pros::Rotation object to use
+         * @param port the signed port of the rotation sensor. Positive if the sensor is reversed, Negative otherwise
          *
          * @b Example:
          * @code {.cpp}
          * void initialize() {
-         *     lemlib::V5RotationSensor encoder = pros::Rotation(1);
+         *     // rotation sensor on port 1, which is reversed
+         *     lemlib::V5RotationSensor encoder(-1);
          * }
          * @endcode
          */
-        V5RotationSensor(pros::Rotation encoder);
+        V5RotationSensor(ReversibleSmartPort port);
         /**
-         * @brief Construct a new V5 Rotation Sensor
+         * @brief Create a new V5 Rotation Sensor
          *
-         * @param port the signed port of the rotation sensor. Positive if the sensor is reversed, Negative otherwise
+         * @param encoder the pros::Rotation object to use
          *
          * @b Example:
          * @code {.cpp}
          * void initialize() {
-         *     // rotation sensor on port 1, which is reversed
-         *     lemlib::V5RotationSensor encoder(-1);
+         *     lemlib::V5RotationSensor encoder = lemlib::V5RotationSensor::from_pros_rot(pros::Rotation(1));
          * }
          * @endcode
          */
-        V5RotationSensor(std::int8_t port);
+        static V5RotationSensor from_pros_rot(pros::Rotation encoder);
```

Comment:
why do we not just overload the constructor?

Reply:
Because it leads to issues with ambiguous calls, since `Motor(1, 200_rpm)` could be either the SmartPort ctor or the pros::Motor ctor